keps/sig-node/5328-node-capabilities/README.md

```diff
@@ -0,0 +1,864 @@
+
+####
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NodeCapabilities
+  - Components depending on the feature gate: kubelet, kube-scheduler, kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Yes. Introducing new capabilities can prevent pod scheduling or pod request admission when the nodes cannot satisfy the pod requirements. 
+*Note:* Enabling the NodeCapabilities feature gate alone doesn't alter default behavior; it only activates the underlying framework. Actual changes to behaviors like pod scheduling or admission happen once nodes start publishing specific capabilities (for instance, Kubelet reporting a feature gate's status as a capability) and control plane components are updated to consume and act on this information.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled, but a specific rollback order must be followed to prevent scheduling disruptions. The feature must be disabled on control plane components before being Disabled on nodes. When the feature is disabled on a Node, the Kubelet would clear the NodeCapabilities field from the Node's status. Existing pods that are already running are unaffected by this process.
```

Comment:
I'm not a fan here, because the typical method for downgrades means that you downgrade nodes first.
So thinking more ahead, enabling the feature by default in beta means that potential we won't be able to do a real rollback that way and will require an additional step.

This is getting to the point that I had above in the version skew section - I think it depends how we'll do that.
If we would rely purely on existing of `nodeCapabilities` capability and ignore them (per node) otherwise, we actually can first rollback the nodes.